spec/helpers/ruby_version_spec.rb

```diff
@@ -84,10 +84,89 @@
             2.4.19
         EOF
 
-        ruby_version   = LanguagePack::RubyVersion.new(@bundler.install.ruby_version, is_new: true)
+        ruby_version   = LanguagePack::RubyVersion.bundle_platform_ruby(bundler_output: @bundler.install.ruby_version)
         version_number = "3.2.3"
         version        = "ruby-#{version_number}"
-        expect(ruby_version.version_without_patchlevel).to eq(version)
+        expect(ruby_version.version_for_download).to eq(version)
+        expect(ruby_version.engine_version).to eq(version_number)
+        expect(ruby_version.engine).to eq(:ruby)
+      end
+    end
+  end
+
+  it "detects RC Ruby from Gemfile.lock" do
+    Hatchet::App.new("default_ruby").in_directory_fork do |_|
```

Comment:
Is the app name `default_ruby` accurate for this test and will it conflict with the test that follows which also uses `default_ruby` as the test name?

Reply:
> will it conflict with the test that follows which also uses default_ruby as the test name?

This `in_directory_fork` moves the app to a tempdir and then runs the code in the block in a forked Ruby process, this gives us process isolation for things like env var modifications, and the tempdir allows for competing disk modifications (without conflict). So this should be perfectly safe.

> Is the app name default_ruby accurate for this test 

In this case the base `default_ruby` app isn't really being used. The body of the test replaces both the Gemfile and Gemfile.lock. I use this app by default (heh) as it's the smallest fixture I've got. I could build up an app from scratch instead of using a fixture in the future.